test/widgets/action_sheet_test.dart

```diff
@@ -1085,6 +1086,139 @@ void main() {
+
+      await tester.pump();
+      await tester.longPress(find.text(someChannel.name).hitTestable());
+      await tester.pump(const Duration(milliseconds: 250));
+    }
+
+    final actionSheetFinder = find.byType(BottomSheet);
+    Finder findButtonForLabel(String label) =>
+      find.descendant(of: actionSheetFinder, matching: find.text(label));
+
+    void checkButton(String label) {
+      check(findButtonForLabel(label)).findsOne();
+    }
+
+    group('showChannelActionSheet', () {
+      void checkButtons() {
+        check(actionSheetFinder).findsOne();
+        checkButton('Mark channel as read');
+      }
+
+      testWidgets('show from inbox', (tester) async {
+        await prepare();
+        await showFromInbox(tester);
+        checkButtons();
+      });
+
+      testWidgets('show from subscription list', (tester) async {
+        await prepare();
+        await showFromSubscriptionList(tester);
+        checkButtons();
+      });
+
+      testWidgets('show with no unread messages', (tester) async {
+        await prepare(unreadMsgs: eg.unreadMsgs());
+        await showFromSubscriptionList(tester);
+        check(actionSheetFinder).findsNothing();
+      });
+    });
+
+    group('MarkChannelAsReadButton', () {
+      void checkRequest(int streamId) {
+        check(connection.takeRequests()).single.isA<http.Request>()
+          ..method.equals('POST')
+          ..url.path.equals('/api/v1/messages/flags/narrow')
+          ..bodyFields.deepEquals({
+            'anchor': 'oldest',
+            'include_anchor': 'false',
+            'num_before': '0',
+            'num_after': '1000',
+            'narrow': jsonEncode([
+              {'operator': 'stream', 'operand': streamId},
+              {'operator': 'is', 'operand': 'unread'}
+            ]),
+            'op': 'add',
+            'flag': 'read',
+          });
+      }
+
+      testWidgets('happy path from inbox', (tester) async {
+        await prepare();
+        await showFromInbox(tester);
+        connection.prepare(json: UpdateMessageFlagsForNarrowResult(
+          processedCount: 1, updatedCount: 1,
+          firstProcessedId: null, lastProcessedId: null,
+          foundOldest: true, foundNewest: true).toJson());
+        await tester.tap(findButtonForLabel('Mark channel as read'));
+        await tester.pumpAndSettle();
+
+        checkNoErrorDialog(tester);
+        checkRequest(someChannel.streamId);
+      });
+
+      testWidgets('request fails', (tester) async {
+          await prepare();
+          await showFromInbox(tester);
+
+          // Prepare error response
+          connection.prepare(httpStatus: 400, json: {
+          'result': 'error', 'code': 'BAD_REQUEST', 'msg': ''});
+
+          // Tap and wait for dialog
+          await tester.tap(findButtonForLabel('Mark channel as read'));
+          await tester.pump(); //
+          await tester.pumpAndSettle(); // Wait for dialog animation
```

Comment:
The `await tester.pump(); //` looks unnecessary, comparing with the corresponding topic action sheet test.